packages/cli/src/controllers/dynamicNodeParameters.controller.ts

```diff
@@ -0,0 +1,153 @@
+	}
+	next();
+};
+
+@Service()
+@Authorized()
+@RestController('/dynamic-node-parameters')
+export class DynamicNodeParametersController {
+	constructor(private readonly service: DynamicNodeParametersService) {}
+
+	@Middleware()
+	parseQueryParams(req: BaseRequest, res: Response, next: NextFunction) {
+		const { credentials, currentNodeParameters, nodeTypeAndVersion } = req.query;
+		if (!nodeTypeAndVersion) {
+			throw new BadRequestError('Parameter nodeTypeAndVersion is required.');
+		}
+		if (!currentNodeParameters) {
+			throw new BadRequestError('Parameter currentNodeParameters is required.');
+		}
+
+		req.params = {
+			nodeTypeAndVersion: jsonParse(nodeTypeAndVersion),
+			currentNodeParameters: jsonParse(currentNodeParameters),
+			credentials: credentials ? jsonParse(credentials) : undefined,
+		};
+
+		next();
+	}
+
+	/** Returns parameter values which normally get loaded from an external API or get generated dynamically */
+	@Get('/options')
+	async getOptions(req: OptionsRequest): Promise<INodePropertyOptions[]> {
+		const { path, methodName, loadOptions } = req.query;
+		const { credentials, currentNodeParameters, nodeTypeAndVersion } = req.params;
+		const additionalData = await getBase(req.user.id, currentNodeParameters);
+
+		if (methodName) {
+			return this.service.getOptionsViaMethodName(
+				methodName,
+				path,
+				additionalData,
+				nodeTypeAndVersion,
+				currentNodeParameters,
+				credentials,
+			);
+		}
+
+		if (loadOptions) {
+			return this.service.getOptionsViaLoadOptions(
+				jsonParse(loadOptions),
+				additionalData,
+				nodeTypeAndVersion,
+				currentNodeParameters,
+				credentials,
+			);
+		}
+
+		return [];
+	}
+
+	@Get('/resource-locator-results', { middlewares: [assertMethodName] })
+	async getResourceLocatorResults(
+		req: ResourceLocatorResultsRequest,
+	): Promise<INodeListSearchResult | undefined> {
+		const { path, methodName, filter, paginationToken } = req.query;
+		const { credentials, currentNodeParameters, nodeTypeAndVersion } = req.params;
+		const additionalData = await getBase(req.user.id, currentNodeParameters);
+		return this.service.getResourceLocatorResults(
+			methodName,
+			path,
+			additionalData,
+			nodeTypeAndVersion,
+			currentNodeParameters,
+			credentials,
+			filter,
+			paginationToken,
+		);
+	}
+
+	@Get('/resource-mapper-fields', { middlewares: [assertMethodName] })
+	async getResourceMappingFields(
+		req: ResourceMapperFieldsRequest,
+	): Promise<ResourceMapperFields | undefined> {
+		const { path, methodName } = req.query;
+		const { credentials, currentNodeParameters, nodeTypeAndVersion } = req.params;
+		const additionalData = await getBase(req.user.id, currentNodeParameters);
+		return this.service.getResourceMappingFields(
+			methodName,
+			path,
+			additionalData,
+			nodeTypeAndVersion,
+			currentNodeParameters,
+			credentials,
+		);
+	}
+}
+
+type BaseRequest<QueryParams = {}> = AuthenticatedRequest<
```

Comment:
Request types are usually namespaced at `requests.ts`.

Reply:
reverted it. but at some point I think we need to figure out a more modular way to manage all these interfaces, instead of adding them all to a single file.